From 0cdc050ea0aa3df82784b31c4475bce01240301d Mon Sep 17 00:00:00 2001
From: Mark Brown <broonie@kernel.org>
Date: Thu, 11 Sep 2014 21:39:05 +0100
Subject: [PATCH] ARM: samsung: Fix mismerge with new libfdt changes

The merge with the new FDT changes incorrectly changed mfc_mem to be a
local variable causing boot failures.

Signed-off-by: Mark Brown <broonie@kernel.org>
---
 arch/arm/plat-samsung/s5p-dev-mfc.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm/plat-samsung/s5p-dev-mfc.c b/arch/arm/plat-samsung/s5p-dev-mfc.c
index 97ece99..a292728 100644
--- a/arch/arm/plat-samsung/s5p-dev-mfc.c
+++ b/arch/arm/plat-samsung/s5p-dev-mfc.c
@@ -127,27 +127,27 @@ int __init s5p_fdt_find_mfc_mem(unsigned long node, const char *uname,
 {
 	const __be32 *prop;
 	int len;
-	struct s5p_mfc_dt_meminfo mfc_mem;
+	struct s5p_mfc_dt_meminfo *mfc_mem = data;
 
 	if (!data)
 		return 0;
 
-	if (!of_flat_dt_is_compatible(node, mfc_mem.compatible))
+	if (!of_flat_dt_is_compatible(node, mfc_mem->compatible))
 		return 0;
 
 	prop = of_get_flat_dt_prop(node, "samsung,mfc-l", &len);
 	if (!prop || (len != 2 * sizeof(unsigned long)))
 		return 0;
 
-	mfc_mem.loff = be32_to_cpu(prop[0]);
-	mfc_mem.lsize = be32_to_cpu(prop[1]);
+	mfc_mem->loff = be32_to_cpu(prop[0]);
+	mfc_mem->lsize = be32_to_cpu(prop[1]);
 
 	prop = of_get_flat_dt_prop(node, "samsung,mfc-r", &len);
 	if (!prop || (len != 2 * sizeof(unsigned long)))
 		return 0;
 
-	mfc_mem.roff = be32_to_cpu(prop[0]);
-	mfc_mem.rsize = be32_to_cpu(prop[1]);
+	mfc_mem->roff = be32_to_cpu(prop[0]);
+	mfc_mem->rsize = be32_to_cpu(prop[1]);
 
 	return 1;
 }
-- 
1.8.0

